lua/lvim/lsp/providers/tailwindcss.lua

```diff
@@ -0,0 +1,20 @@
+local utils = require "lvim.utils"
+
+-- lsp attaches only if you have tailwindcss installed and a config file named
+-- tailwind.config.js or tailwind.config.cjs.
+local project_has_tailwindcss_dependency = function()
+  return (vim.fn.glob "tailwind.config.*js" ~= "" and utils.is_in_package_json "tailwindcss")
+end
+
+local opts = {
+  root_dir = function(fname)
+    if not project_has_tailwindcss_dependency() then
+      return
+    end
+
+    local default_root_dir = require("lspconfig").tailwindcss.document_config.default_config.root_dir
+    return default_root_dir(fname)
+  end,
+}
```

Comment:
```suggestion
local opts = {
  root_dir = function(fname)
     return util.root_pattern('tailwind.config.js', 'tailwind.config.ts')(fname)
        or util.root_pattern('postcss.config.js', 'postcss.config.ts')(fname)
  end,
}
```

Reply:
so this would enough for the PR regarding tailwind

Reply:
I don't agree...

Does postcss implies tailwind? (don't think so, but I don't know).

Also, doesn't handle '.cjs'. For reference, https://github.com/tailwindlabs/tailwindcss-intellisense/blob/master/packages/vscode-tailwindcss/src/extension.ts#L50.

And I haven't seen any '.ts' mentioned on tailwind docs (or the vscode ext, but I haven't checked it thoroughly), even for a typescript project.

From what I've tested, I think it's ok to not check if tailwind is installed in the project.

I'm deciding between
`
util.root_pattern("tailwind.config.js", "tailwind.config.cjs", "tailwind.js", "tailwind.cjs")(fname)
`
or
`util.root_pattern "tailwind.*js" (fname)`, what do you think?

Reply:
> Does postcss implies tailwind? (don't think so, but I don't know).

I don't know either 😄 

if it's not, then maybe we should fix that in lspconfig itself, see https://github.com/neovim/nvim-lspconfig/blob/80fddd28de5fe0642da7adee955f5762b933e730/lua/lspconfig/server_configurations/tailwindcss.lua#L106-L107

Reply:
The usage of postcss depends upon the framework being used and how one set up the project.
The `tailwindcss` cli,  does not use add postcss in the package.json file.

Reply:
@lvimuser, apply this to fix the tests
```diff
diff --git a/tests/specs/lsp_spec.lua b/tests/specs/lsp_spec.lua
index 7d9a338..01e5e1d 100644
--- a/tests/specs/lsp_spec.lua
+++ b/tests/specs/lsp_spec.lua
@@ -53,10 +53,14 @@ a.describe("lsp workflow", function()
   a.it("should only include one server per generated template", function()
     require("lvim.lsp").setup()
 
+    local allowed_dupes = { "tailwindcss" }
     for _, file in ipairs(vim.fn.glob(lvim.lsp.templates_dir .. "/*.lua", 1, 1)) do
       local content = {}
       for entry in io.lines(file) do
-        table.insert(content, entry)
+        local server_name = entry:match [[.*setup%("(.*)"%)]]
+        if not vim.tbl_contains(allowed_dupes, server_name) then
+          table.insert(content, server_name)
+        end
       end
       local err_msg = ""
       if #content > 1 then
```